src/conn/conn_oligarch.c

```diff
@@ -15,30 +15,29 @@
 static int
 __disagg_pick_up_checkpoint(WT_SESSION_IMPL *session, uint64_t checkpoint_id)
 {
+    WT_CONFIG_ITEM cval;
     WT_CONNECTION_IMPL *conn;
-    WT_CURSOR *md_cursor;
+    WT_CURSOR *cursor, *md_cursor;
     WT_DECL_ITEM(item);
     WT_DECL_RET;
-    WT_SESSION_IMPL *internal_session;
-    char buf[4096], *cfg_ret,
-      *metadata_value_cfg; /* TODO the 4096 puts an upper bound on metadata entry length */
-    const char *cfg[3], *current_value, *metadata_key, *metadata_value;
-    size_t len, sep;
+    WT_SESSION_IMPL *internal_session, *shared_metadata_session;
+    size_t len, metadata_value_cfg_len;
     uint64_t global_checkpoint_id;
+    char *buf, *cfg_ret, *metadata_value_cfg;
+    const char *cfg[3], *current_value, *metadata_key, *metadata_value;
 
     conn = S2C(session);
 
+    buf = NULL;
+    cursor = NULL;
     internal_session = NULL;
     md_cursor = NULL;
     metadata_key = NULL;
     metadata_value = NULL;
     metadata_value_cfg = NULL;
+    shared_metadata_session = NULL;
 
-    WT_ERR(__wt_scr_alloc(session, 4096, &item));
-
-    /* Only followers are allowed to pick up new checkpoints. */
-    if (conn->oligarch_manager.leader)
-        WT_ERR(EINVAL);
```

Comment:
I removed this because actually this function can actually be called from the leader during step up (if the call to reconfig specifies both the checkpoint ID and the role change).